ckanext/dcat/harvesters/configuration_processors.py

```diff
@@ -0,0 +1,329 @@
+import re
+from abc import ABCMeta, abstractmethod
+from collections import OrderedDict
+
+from ckan import model
+from ckan import plugins as p
+from ckan.lib.munge import substitute_ascii_equivalents
+from ckan.logic import NotFound, get_action
+
+
+def munge_to_length(string, min_length, max_length):
+    '''Pad/truncates a string'''
+    if len(string) < min_length:
+        string += '_' * (min_length - len(string))
+    if len(string) > max_length:
+        string = string[:max_length]
+    return string
+
+
+def munge_tag(tag):
+    tag = substitute_ascii_equivalents(tag)
+    tag = tag.lower().strip()
+    tag = re.sub(r'[^a-zA-Z0-9\- ]', '', tag)
+    tag = munge_to_length(tag, model.MIN_TAG_LENGTH, model.MAX_TAG_LENGTH)
+    return tag
+
+
+def get_extra(key, package_dict):
+    for extra in package_dict.get('extras', []):
+        if extra['key'] == key:
+            return extra
+
+
+class BaseConfigProcessor:
+    __metaclass__ = ABCMeta
+
+    @staticmethod
+    @abstractmethod
+    def check_config(config_obj):
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def modify_package_dict(package_dict, config, dcat_dict):
+        raise NotImplementedError
+
+
+class ResourceOrder(BaseConfigProcessor):
+
+    @staticmethod
+    def check_config(config_obj):
+        if 'resources_order' in config_obj:
+            if not isinstance(config_obj['resources_order'], list):
+                raise ValueError('Resources order should be provided as sting with')
```

Comment:
I think the error message should be `Resource order should be provided as a list of strings`